client/secureclient/src/main/java/com/linkedin/openhouse/client/ssl/WebClientFactory.java

```diff
@@ -61,13 +65,17 @@ protected WebClient createWebClient(
       @NonNull String baseUrl, String token, String truststoreLocation)
       throws MalformedURLException, SSLException {
     String transportProtocol = getTransportProtocol(baseUrl);
+    HttpClient httpClient = null;
     if (HTTPS.equals(transportProtocol)) {
-      return createSecureWebClient(baseUrl, truststoreLocation);
+      httpClient = createSecureHttpClient(truststoreLocation);
     } else if (HTTP.equals(transportProtocol)) {
-      return createWebClient(baseUrl);
+      httpClient = createHttpClient(baseUrl);
     } else {
       throw new RuntimeException("The transport protocol must be https/http");
     }
+    return getWebClientBuilder(baseUrl)
```

Comment:
```suggestion
    return buildWebClient(baseUrl)
```
We can move the connector part to buildWebClient itself.